skrub/_agg_joiner.py

```diff
@@ -130,119 +209,73 @@ class AggJoiner(TransformerMixin, BaseEstimator):
-                The auxiliary table.
-
-        Returns
-        -------
-        X, aux_table: DataFrameLike
-            The validated main and auxiliary dataframes.
-        """
-        # Polars lazyframes will raise an error here.
-        if not hasattr(X, "__dataframe__"):
-            raise TypeError(f"'X' must be a dataframe, got {type(X)}.")
-        if isinstance(aux_table, str):
-            if aux_table == "X":
-                return X, X
-            raise ValueError("'aux_table' must be a dataframe or the string 'X'.")
-        elif not hasattr(aux_table, "__dataframe__"):
-            raise TypeError(
-                "'aux_table' must be a dataframe or the string 'X', got"
-                f" {type(aux_table)}. If you have more than one 'aux_table',"
-                " use the MultiAggJoiner instead."
-            )
-
-        if (is_pandas(X) and not is_pandas(aux_table)) or (
-            is_polars(X) and not is_polars(aux_table)
-        ):
-            raise TypeError(
-                "'X' and 'aux_table' must be of the same dataframe type, got"
-                f"{type(X)} and {type(aux_table)}"
-            )
-
-        return X, aux_table
-
     def _check_inputs(self, X):
         """Check inputs before fitting.
 
         Parameters
         ----------
         X : DataFrameLike
-            Input data, based table on which to left join the
+            Input data, base table on which to left join the
             auxiliary table.
         """
-        X, self._aux_table = self._check_dataframes(X, self.aux_table)
-
         self._main_key, self._aux_key = _join_utils.check_key(
             self.main_key, self.aux_key, self.key
         )
         _join_utils.check_missing_columns(X, self._main_key, "'X' (the main table)")
         _join_utils.check_missing_columns(self._aux_table, self._aux_key, "'aux_table'")
+        _join_utils.check_missing_columns(
+            self._aux_table, s.make_selector(self.cols), ""
+        )
 
         # If no `cols` provided, all columns but `aux_key` are used.
-        if self.cols is None:
-            self._cols = list(set(self._aux_table.columns) - set(self._aux_key))
-        elif isinstance(self.cols, str):
-            self._cols = [
-                self.cols,
-            ]
-        else:
-            self._cols = self.cols
-        _join_utils.check_missing_columns(self._aux_table, self._cols, "'aux_table'")
-
-        if self.operations is None:
-            self._operations = ["mean", "mode"]
-        elif isinstance(self.operations, str):
-            self._operations = [
-                self.operations,
-            ]
-        else:
-            self._operations = self.operations
+        self._cols = s.make_selector(self.cols) - self._aux_key
 
-        self.num_operations, self.categ_operations = split_num_categ_operations(
-            self._operations
-        )
+        self._operations = atleast_1d_or_none(self.operations)
```

Comment:
I think operations cannot be None

Reply:
You're right ! Removing this for the `main_key` check also 

thanks :)